python/pypto/language/distributed/op/tensor_ops.py

```
@@ -519,6 +535,28 @@ def allreduce(
         pub = pld.tensor.allreduce(pub, sig, op=pld.ReduceOp.Sum)
         pub = pld.tensor.allreduce(pub, sig, op=pld.ReduceOp.Sum, mode="ring")
 
+        # Mesh mode on the host orchestrator can omit the signal argument:
+        pub = pld.tensor.allreduce(pub, op=pld.ReduceOp.Sum)
+
+    **Signal shapes:**
+    Mesh: ``[NR, 1]`` — one global barrier covers all peer reads.
+    Ring: ``[2 × (NR − 1), NR]`` — one fresh row per ring step.
```

Ruff (0.15.21)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)